packages/ngtools/webpack/src/transformers/remove-ivy-dev-calls.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+import { collectDeepNodes } from './ast_helpers';
+import { RemoveNodeOperation, StandardTransform, TransformOperation } from './interfaces';
+import { makeTransform } from './make_transform';
+
+export function removeClassMetadata(): ts.TransformerFactory<ts.SourceFile> {
```

Comment:
there is a ton of duplication in removeClassMetadata and removeNgModuleScope functions. Not only that we are adding unnecessary code to our code base, but we are also making the transform twice as slow because we iterate over all the nodes twice.

I suggest you create a single transformer that will remove both types of nodes. You can still have two separate flags to control the behavior of the transformer, but you don't need two transformers.